features/steps/steps.go

```diff
@@ -218,3 +236,230 @@ func (c *Component) theFollowingExportStartEventsAreProduced(events *godog.Table
+	return dataset.Version{
+		ID:         ds.ID + "UUID",
+		Edition:    "latest",
+		Version:    1,
+		State:      "published",
+		IsBasedOn:  ds.IsBasedOn,
+		Dimensions: dims.Items,
+		Links: dataset.Links{
+			Dataset: ds.Links.Self,
+			Version: dataset.Link{
+				ID:  "1",
+				URL: fmt.Sprintf("%s/editions/%s/versions/%d", ds.Links.Self.URL, "latest", 1),
+			},
+			Self: dataset.Link{
+				ID:  "1",
+				URL: fmt.Sprintf("%s/editions/%s/versions/%d", ds.Links.Self.URL, "latest", 1),
+			},
+		},
+	}
+}
+
+func (c *Component) makeMetadata(v dataset.Version) interface{} {
+	// logic referenced from dp-dataset-api/metadata/models/metadata.go CreateCantabularMetaDataDoc() (commit hash dd295b15704d87c0cb08d127ca108ce2f8df05a7)
+	// no metadata links are being set at present in this function, but it is envisaged the links will be set at some point, so I am setting a basic link in
+	//this test. The returned value is an extremely cutdown version of dp-dataset-api/metadata/models/Metadata
+
+	return struct {
+		Links interface{} `json:"links,omitempty"`
+	}{Links: struct {
+		Self interface{} `json:"self,omitempty"`
+	}{Self: struct {
+		HRef string `json:"href,omitempty"`
+		ID   string `json:"id,omitempty"`
+	}{HRef: v.Links.Self.URL + "/metadata"}}}
+}
+
+func (c *Component) makeDimensionOptions(v dataset.Version, dims []cantabular.Dimension) []dataset.Option {
+	if len(v.Dimensions) != len(dims) {
+		c.Fatalf("dimension lists  not of same length in Component::makeDimensionOptions")
+	}
+	sort.Slice(v.Dimensions, func(i, j int) bool { return v.Dimensions[i].ID < v.Dimensions[j].ID })
+	sort.Slice(dims, func(i, j int) bool { return dims[i].Variable.Name < dims[j].Variable.Name })
+
+	var dimOpts []dataset.Option
+	for i, d := range v.Dimensions {
+
+		var opts []dataset.Option
+		for _, o := range dims[i].Categories {
+			opts = append(opts, dataset.Option{
+				DimensionID: d.ID,
+				Label:       o.Label,
+				Option:      o.Code,
+				Links: dataset.Links{
+					Versions: v.Links.Version,
+					CodeList: d.Links.CodeList,
+					Code: dataset.Link{
+						ID:  o.Code,
+						URL: fmt.Sprintf("%s/codes/%s", d.Links.CodeList.URL, o.Code),
+					},
+				},
+			})
+		}
+
+		optsj, err := json.Marshal(dataset.Options{Items: opts, Count: len(opts), TotalCount: len(opts)})
+		if err != nil {
+			c.Fatalf("error marshalling dimension options in Component::theFollowingRecipeIsUsed: %v", err)
+		}
+		if err = c.DatasetFeature.theFollowingOptionsDocumentIsAvailable(v.Links.Dataset.ID, v.Edition, v.Links.Version.ID, d.ID, &godog.DocString{Content: string(optsj)}); err != nil {
+			c.Fatalf("error setting dimensions into DatasetFeature in Component::theFollowingRecipeIsUsed: %v", err)
+		}
+
+		dimOpts = append(dimOpts, opts...)
+	}
+
+	return dimOpts
+}
+
+func (c *Component) theDatasetJSONResult(expected *godog.DocString) error {
+	var got, expt api.GetDatasetJSONResponse
```

Comment:
What's wrong with the built in `I should receive the following JSON response` step?

Reply:
This was to provide some extras over what the `I should receive the following JSON response` does. Specifically because the tests use a fake cantabular server, I want to ignore the hostname embedded in certain urls, and this function does that.
This concern might disappear depending on what we do with the next issue below 👇

Reply:
I have changed this to use the standard step: `I should receive the following JSON response`